File: ios/Classes/Utils/StatusMessages.swift
//
//  StatusMessages.swift
//  rppg_common
//
//  Created by Wegile on 31/01/24.
//

import Foundation
import RPPGCommon

/// Custom response pair
class StatusMessages {
    static func getMessage(forStatus status: RPPGSocketMessageDataStatus.StatusCode?) -> (statusCode: String, statusMessage: String) {
        switch status {
        case .success:
            return ("success", "Calculating vitals…")

        case .noFace:
            return ("noFace", "No Face Detected")

        case .faceLost:
            return ("faceLost", "Please make sure your face is still in the frame")

        case .calibrating:
            return ("calibrating", "Analyzing PPG signals…")

        case .recalibrating:
            return ("recalibrating", "Recalibrating…")

        case .brightLightIssue:
            return ("brightLightIssue", "Bright light interference")

        case .noiseDuringExecution:
            return ("noiseDuringExecution", "Interference detected")
            
        @unknown default:
            return ("unknown", "")
        }
    }
}

File: android/src/main/kotlin/com/example/rppg_common/utils/StatusMessages.kt
package com.example.rppg_common.utils

import com.rppg.library.common.socket.model.MessageStatus

/// Custom response pair
class StatusMessages {
    companion object {
        fun getMessage(status: MessageStatus): Pair<String, String> {
            return when (status) {
                MessageStatus.SUCCESS -> "success" to "Calculating vitals…"
                MessageStatus.NO_FACE -> "noFace" to "No Face Detected"
                MessageStatus.FACE_LOST -> "faceLost" to "Please make sure your face is still in the frame"
                MessageStatus.CALIBRATING -> "calibrating" to "Analyzing PPG signals…"
                MessageStatus.RECALIBRATING -> "recalibrating" to "Recalibrating…"
                MessageStatus.BRIGHT_LIGHT_ISSUE -> "brightLightIssue" to "Bright light interference"
                MessageStatus.NOISE_DURING_EXECUTION -> "noiseDuringExecution" to "Interference detected"
                else -> "unknown" to ""
            }
        }
    }
}
